.github/actions/update-stars/main.cjs

```diff
@@ -0,0 +1,108 @@
+const core = require('@actions/core');
+const {readFileSync, writeFileSync} = require("fs");
+const axios = require("axios");
+
+const ghGraphQlUrl = 'https://api.github.com/graphql'
+const gitlabGraphQlUrl = 'https://gitlab.com/api/graphql'
+const githubNameRegexp = new RegExp('https://github.com/([a-zA-Z0-9][a-zA-Z0-9-]{0,38}/[a-zA-Z0-9._-]{1,100})')
+const gitlabNameRegExp = new RegExp('https://gitlab.com/([\\w-]+/[\\w-]+)')
+
+async function doGraphQlQuery(url, query, headers = {}) {
+    try {
+        let fetchResponse = await axios({
```

Comment:
shouldn't need to use `axios`. can just use the `fetch` built into node - or possibly `undici` if on an older version of node, which is the `fetch` that is bundled with node

Reply:
Sadly, GitHub action only allows Node 12 and Node 16.
(And the doc don't even mention Node 12).

I switch to `node-fetch`, which is more close to the native `fetch` (less work when Node 18 will be available)

Reply:
I think you just need to add https://github.com/actions/setup-node. E.g. in https://github.com/svelte-society/sveltesociety.dev/pull/424 you used Node 18

Reply:
A workflow can use `setup-node` (or whatever they need, as they run on a full container), but an action is a bit more limited:

- https://docs.github.com/en/actions/creating-actions/creating-a-javascript-action
- https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsusing-for-javascript-actions

---

Maybe there are other method to make an action, or I can transform the action into a workflow step (If everything is possible as a workflow step) :thinking:

Reply:
Ah, okay. In that case I think we should use `undici` rather than `node-fetch`. It's exactly the same as what's included in Node so later when we use a newer Node we can just drop the dependency and everything will keep working exactly the same